src/v/redpanda/application.cc

```diff
@@ -689,6 +689,7 @@ void application::wire_up_services() {
 
 void application::wire_up_redpanda_services() {
     ss::smp::invoke_on_all([] {
+        resources::available_memory::local().register_metrics();
```

Comment:
@dotnwat so this is the "service" I was thinking about when I asked about wiring things up on slack.

Ultimately, I did not make it a `sharded<foo>` at all but just followed kind of the model of the `chunk_cache`. Actually, it is "just a `thread_local` object" which works without any setup, except that you need to call `register_metrics` once to do the registration (but this would not happen e.g., when you are creating and destroying batch_cache in a test).

Reply:
got it. thanks for the explanation